File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/NumpadManager.java
package com.remotearthsolutions.expensetracker.utils;

import android.view.View;
import android.widget.EditText;
import com.remotearthsolutions.expensetracker.fragments.NumpadFragment;

public class NumpadManager implements NumpadFragment.Listener {

    private EditText displayEdtxt;
    private boolean isOperationDone;
    private double result = 0;
    private String lastOperation, lastValue = "0";

    public void attachDisplay(EditText inputDigitEdtxt) {
        this.displayEdtxt = inputDigitEdtxt;
    }

    public void attachDeleteButton(View deleteButton) {
        deleteButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int len = displayEdtxt.getText().toString().length();
                if (len > 0 && !isOperationDone) {
                    displayEdtxt.setText(displayEdtxt.getText().toString().substring(0, len - 1));
                }
            }
        });
    }

    @Override
    public void onNumpadButtonClick(String value) {
        if (isOperationDone) {
            displayEdtxt.setText("");
            isOperationDone = false;
        }

        String str = displayEdtxt.getText().toString();
        if (value.equals(".") && str.contains(".")) {
            displayEdtxt.setText(str);
        } else {
            str += value;
            displayEdtxt.setText(str);
        }
    }

    @Override
    public void onMathOperationButtonClick(String operation) {

        if (isOperationDone && lastOperation != null) {
            lastOperation = operation;
            return;
        }

        String val = displayEdtxt.getText().toString();
        if (val.length() == 0 || val.equals(".")) {
            return;
        }

        double currentVal = Double.parseDouble(val);
        double lastVal = Double.parseDouble(lastValue);

        if (lastOperation == null) {
            result = currentVal;
        } else {
            if (lastOperation.equals("+")) {
                result = lastVal + currentVal;
            } else if (lastOperation.equals("-")) {
                result = lastVal - currentVal;
            } else if (lastOperation.equals("*")) {
                result = lastVal * currentVal;
            } else if (lastOperation.equals("/")) {
                result = lastVal / currentVal;
            }
        }

        if (operation.equals("=")) {
            lastOperation = null;
            lastValue = "0";
        } else {
            lastOperation = operation;
            lastValue = Double.toString(result);
        }

        displayEdtxt.setText(Double.toString(result));
        isOperationDone = true;
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/utils/NumpadManager.kt
package com.remotearthsolutions.expensetracker.utils

import android.view.View
import android.widget.EditText
import com.remotearthsolutions.expensetracker.fragments.NumpadFragment
import com.remotearthsolutions.expensetracker.utils.Utils.formatDecimalValues

class NumpadManager : NumpadFragment.Listener {
    private var displayEdtxt: EditText? = null
    private var isOperationDone = false
    private var result = 0.0
    private var lastOperation: String? = null
    private var lastValue = "0"
    fun attachDisplay(inputDigitEdtxt: EditText?) {
        displayEdtxt = inputDigitEdtxt
    }

    fun attachDeleteButton(deleteButton: View) {
        deleteButton.setOnClickListener {
            val len = displayEdtxt!!.text.toString().length
            if (len > 0 && !isOperationDone) {
                displayEdtxt!!.setText(displayEdtxt!!.text.toString().substring(0, len - 1))
            }
        }
    }

    override fun onNumpadButtonClick(value: String?) {
        if (isOperationDone) {
            displayEdtxt!!.setText("")
            isOperationDone = false
        }
        var str = displayEdtxt!!.text.toString()
        if (value == "." && str.contains(".")) {
            displayEdtxt!!.setText(str)
        } else {
            str += value
            displayEdtxt!!.setText(str)
        }
    }

    @Throws(NumberFormatException::class)
    override fun onMathOperationButtonClick(operation: String?) {
        if (isOperationDone && lastOperation != null) {
            lastOperation = operation
            return
        }
        val `val` = displayEdtxt!!.text.toString()
        if (`val`.isEmpty() || `val` == ".") {
            return
        }
        val currentVal: Double
        currentVal = try {
            `val`.toDouble()
        } catch (e: NumberFormatException) {
            throw e
        }
        val lastVal = lastValue.toDouble()
        if (lastOperation == null) {
            result = currentVal
        } else {
            when (lastOperation) {
                "+" -> {
                    result = lastVal + currentVal
                }
                "-" -> {
                    result = lastVal - currentVal
                }
                "*" -> {
                    result = lastVal * currentVal
                }
                "/" -> {
                    result = lastVal / currentVal
                }
            }
        }
        if (operation == "=") {
            lastOperation = null
            lastValue = "0"
        } else {
            lastOperation = operation
            lastValue = result.toString()
        }
        displayEdtxt!!.setText(
            formatDecimalValues(
                result
            )
        )
        isOperationDone = true
    }
}